lib/xcodeproj/project/object/helpers/sort_helper.rb

```diff
@@ -0,0 +1,34 @@
+module Xcodeproj
+  # Wrapper for a string that sorts by name like Xcode does.
+  # @example
+  #   arrayOfFilenames.sort_by { |s| XcodeSortString.new(s) }
+  class XcodeSortString
+    include Comparable
+    attr_reader :str, :scrubbed, :ints_and_strings, :i_s_pattern
+
+    def initialize(str)
+      @str = str
+      @scrubbed = str.downcase
+      # `Integer`+`rescue`: credit to https://stackoverflow.com/a/39025521
+      @ints_and_strings = @scrubbed.scan(/\d+|\D+/).map { |s| begin Integer(s, 10); rescue ArgumentError; s end }
```

Comment:
Can we do this without having to swallow exceptions?

Reply:
Yes, absolutely, by adding a regex match: `{ |s| \d\ === s ? Integer(s, 10) : s }`.
I do not know which one is the most performant, but I thought an exception may be faster than a regex match.
I'll change it.